models/DataObject/Concrete/Dao.php

```diff
@@ -208,7 +207,7 @@ public function update(bool $isUpdate = null): void
             $untouchables = "'" . implode("','", $untouchable) . "'";
             $condition = Helper::quoteInto($this->db, 'src_id = ? AND fieldname not in (' . $untouchables . ") AND ownertype = 'object'", $this->model->getId());
         } else {
-            $condition = 'src_id = ' . $db->quote($this->model->getId()) . ' AND ownertype = "object"';
+            $condition = 'src_id = ' . $this->model->getId() ?: 0 . ' AND ownertype = "object"';
```

Comment:
We need brackets here. Otherwise it will first concat the string.
```suggestion
            $condition = 'src_id = ' . ($this->model->getId() ?? 0) . ' AND ownertype = "object"';
```

Reply:
ahh, thank you for the heads-up

I think we can avoid the null check there according to stan, or is it a false positive?
https://github.com/pimcore/pimcore/actions/runs/9172296100/job/25218475250

Reply:
@kingjia90 It is always true because it was a string with "src_id = " prefix